Declare ghc-toolchain-bin's license

Every other package in the GHC tree carries `license: BSD-3-Clause`; these
two are simply missing the field, so anything reading the `.cabal` -- cabal2nix
among them -- has to be told the license out of band.

--- a/ghc-toolchain-bin.cabal
+++ b/ghc-toolchain-bin.cabal
@@ -6,6 +6,7 @@
 bug-reports:        https://gitlab.haskell.org/ghc/ghc/-/issues
 author:             Ben Gamari
 maintainer:         ben@well-typed.com
+license:            BSD-3-Clause
 copyright:          (c) The GHC Developers

 executable ghc-toolchain-bin
